src/shogun/machine/Composite.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+#include <memory>
+
+#include <shogun/machine/Composite.h>
```

Comment:
I wonder whether the class should be called `Ensemble` ? Or is that inappropriate?

Reply:
Shouldn't there be a `EnsembleMachine` and then `Composite` becomes a helper class in `Pipeline` to have the interface `with` and `then` to add things to the ensemble?